File: app/src/main/java/harkor/mycryptocurrency/model/Cryptocurrency.java
package harkor.mycryptocurrency.model;

public class Cryptocurrency {
    public int id;
    public String tag;
    public double amount;
    public String date;
    public double priceUsd;
    public double priceEur;
    public double pricePln;
    public double priceBtc;

    public Cryptocurrency(int id, String tag, double amount, String date, double priceUsd, double priceEur, double pricePln,double priceBtc) {
        this.id = id;
        this.tag = tag;
        this.amount = amount;
        this.date = date;
        this.priceUsd = priceUsd;
        this.priceEur = priceEur;
        this.pricePln = pricePln;
        this.priceBtc = priceBtc;
    }

    public Cryptocurrency(int id, String tag, double amount) {
        this.id = id;
        this.tag = tag;
        this.amount = amount;
    }
}

File: app/src/main/java/harkor/mycryptocurrency/model/Cryptocurrency.kt
package harkor.mycryptocurrency.model


import com.thoughtbot.expandablerecyclerview.models.ExpandableGroup
import harkor.mycryptocurrency.CryptocurrencyOwnedEntity
import harkor.mycryptocurrency.Details

class Cryptocurrency(title: String, amount: Double?, items: List<Details>) : ExpandableGroup<Details>(title, items)

data class ActualPrices(var  usd: Double=0.0, var eur: Double=0.0, var pln: Double=0.0, var btc: Double=0.0)
data class CryptoFullInfo(val cryptoDbInfo: CryptocurrencyOwnedEntity, var actualPrices: ActualPrices, var expanded:Boolean=false )

object CurrencyCalc {
    fun addSign(amount:String, currencyID:Int):String{
        return when(currencyID){
            1-> "$$amount"
            2-> "${amount}€"
            3-> "${amount}zł"
            else-> "${amount}BTC"
        }
    }
}